### PR TITLE
Fix specs for parse to allow AmbiguousDateTime

### PR DESCRIPTION
**Pull request by billperegoy**
### Summary of changes
When issue:
https://github.com/bitwalker/timex/issues/674

was addressed to allow the parse function to return AmbiguousDateTime,
the specs were not updated and this can cause dialyzer errors in code
that uses this function.

### Checklist

- [x] New functions have typespecs, changed functions were updated
- [x] Same for documentation, including moduledocs
- [x] Tests were added or updated to cover changes
- [x] Commits were squashed into a single coherent commit
- [x] Notes added to CHANGELOG file which describe changes at a high-level

### CODE REVIEW

File: lib/timex.ex [hunk 1]
```diff
@@ -1060,7 +1072,11 @@ defmodule Timex do
 
   and the result will be an integer value of those units or a Duration.
   """
-  @spec diff(Time.t() | Comparable.comparable(), Time.t() | Comparable.comparable(), Comparable.granularity()) ::
+  @spec diff(
+          Time.t() | Comparable.comparable(),
+          Time.t() | Comparable.comparable(),
+          Comparable.granularity()
+        ) ::
```
Comment (billperegoy) [inline lib/timex.ex:1079]: These are unrelated changes that `mix format` made.

**Review by bitwalker — APPROVED** [review]
